ui/imports/shared/stores/TransactionStore.qml

```diff
@@ -105,6 +107,55 @@ QtObject {
         return {}
     }
 
+    function getCollectible(uid) {
+        for(var i=0; i< collectiblesModel.count;i++) {
+            if(uid === collectiblesModel.rowData(i, "uid"))
+                return {
+                    uid: collectiblesModel.rowData(i, "uid"),
+                    chainId: collectiblesModel.rowData(i, "chainId"),
+                    contractAddress: collectiblesModel.rowData(i, "contractAddress"),
+                    tokenId: collectiblesModel.rowData(i, "tokenId"),
+                    name: collectiblesModel.rowData(i, "name"),
+                    imageUrl: collectiblesModel.rowData(i, "imageUrl"),
+                    collectionUid: collectiblesModel.rowData(i, "collectionUid"),
+                    collectionName: collectiblesModel.rowData(i, "collectionName"),
+                }
```

Comment:
Please have a look at `ModelUtils`, we have helper functions to facilitate these operations already :)

Reply:
oh man, that's incredibly handy. Thanks!